multiversx_sdk/core/transactions_factories/smart_contract_transactions_factory.py

```diff
@@ -93,8 +110,10 @@ def create_transaction_for_execute(self,
                 receiver=receiver, transfers=token_transfers)
             receiver = sender
 
+        prepared_arguments = self._encode_execute_arguments(function, list(arguments))
+
         data_parts.append(function) if not data_parts else data_parts.append(arg_to_string(function))
-        data_parts.extend(args_to_strings(arguments))
+        data_parts.extend([arg.hex() for arg in prepared_arguments])
```

Comment:
could have been:
```py
data_parts += [arg.hex() for arg in prepared_arguments]
```
just to stay consistent with the other methods, but it's not a big deal, can stay as it is

Reply:
Will be applied in the very next PR.

Reply:
Fixed here: https://github.com/multiversx/mx-sdk-py/pull/57.